client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -317,23 +317,31 @@ class PreviewFrame extends React.Component {
 
   renderSketch() {
     const doc = this.iframeElement;
+    const localFiles = this.injectLocalFiles();
     if (this.props.isPlaying) {
-      srcDoc.set(doc, this.injectLocalFiles());
+      srcDoc.set(doc, localFiles);
       if (this.props.endSketchRefresh) {
         this.props.endSketchRefresh();
       }
     } else {
       doc.srcdoc = '';
       srcDoc.set(doc, '  ');
     }
+    const matches = localFiles.match(/createCanvas\((\d+), (\d+)\)/);
+    doc.width = `${matches[1]}px`;
+    doc.height = `${matches[2]}px`;
   }
 
   render() {
     return (
       <iframe
+        id="canvas_frame"
         className="preview-frame"
+        width="720px" // TODO
+        height="400px" // TODO
         aria-label="sketch output"
         role="main"
+        scrolling="no"
```

Comment:
there's maybe an (edge) case where we'd want to leave scrolling in—for example if someone made a purposefully very large sketch.

Reply:
I tried, but this kind of yield weird results, since the iframe is slightly bigger than the canvas it encloses. That means that either I allow an error margin when computing the size of the iframe from the size of the canvas (but then the canvas will be slightly off-center) or the scroll bars are there permanently. Neither seem like good solutions. Do you have another idea ?

Reply:
i think this issue is fixed by #685—let me look into merging it!

Reply:
could you also remove this stuff that tries to fix #680? i want to keep this PR contained to #692 if possible.

Reply:
Right, I'm on it. Turns out my code was pretty dirty and non standard on that point. I'm correcting it right now ^^